Stats

LCschool opened the pull request:
# Feature Addition
Closes: #239 
Closes: #319
## Summary

The stats page is attached to the report tab, The top 3 graphs are changeable between different distribution statics and are defaulted to pie charts whereas the bottom two are incident type and descriptors sorted by type. 

## Rationale

Stats page allows the admin to easily view visual data and customize what data they are seeing


## Changes

In addition to the stats page,
incident seeding was modified so that descriptor, and type seed a variety of descriptors and roles,
stats controller was dramatically changed to send distribution data to the page rather than all incidents

## Impact

None

## Testing

Testing checks to see if all props are being sent to the stat page correctly and that only the admin can view

## Screenshots/Video

![example2](https://github.com/user-attachments/assets/2e93d2e7-8a12-4074-8404-0a9be45370ee)

## Checklist

- [x] Code follows the project's coding standards

- [x] Unit tests covering the new feature have been added

- [x] All existing tests pass

- [x] The documentation has been updated to reflect the new feature

## Additional Notes

Room for time series, for use in scatterplots and line graphs to show trends overtime

LCschool committed:
Began blocking out charts and framing out data

LCschool committed:
populated dataframe

LCschool committed:
Graphs are displaying data and dropdown menu is in place

LCschool committed:
Menu is working, position needed

OsmiumCoder committed:
Merge branch 'staging' into stats

OsmiumCoder committed:
tailwind css fix

LCschool committed:
Charts can toggle between pie and bar, bottom two are default set to Type and Descriptor.

LCschool committed:
Wrote Tests

LCschool committed:
fix code style

welshy557 committed:
Merge branch 'staging' into stats

welshy557 committed:
Fixes TS errors and updates chart button hover to lighter gray

welshy557 commented on package.json:
```diff
@@ -29,6 +29,8 @@
         "@types/underscore": "^1.13.0",
         "@vitejs/plugin-react": "^4.3.4",
+        "chart.js": "^4.4.8",
+        "chartjs": "^0.3.24",
```
Don't think we need both of these? chart.js is the correct one I think.

welshy557 committed:
Renames Graph prop interface to GraphProps

welshy557 commented on resources/js/Pages/Report/Stats.tsx:
```diff
@@ -1,5 +1,166 @@
+
+    return (
+        <AuthenticatedLayout>
+            <div className="mx-8 grid grid-cols-1 gap-5 lg:grid-cols-6">
```
Add `<Head title='Statistics' />` here

welshy557 commented on packages.txt:
Remove

welshy557 commented on tsconfig.json:
```diff
@@ -114,5 +114,9 @@
+    "include": [
+      "resources/js/**/*.ts",
+      "resources/js/**/*.tsx",
+      "node_modules/eslint-plugin-react/lib/types.d.ts"
```
Don't think we need `"node_modules/eslint-plugin-react/lib/types.d.ts"`. Can be removed.

welshy557 committed:
Adds select-none to labeled checkbox label

LCschool committed:
Make requested changes

LCschool committed:
fix code style

LCschool committed:
Fixed small labeling issue with binary values

LCschool committed:
Merge remote-tracking branch 'origin/stats' into stats

welshy557 committed:
Updates report field from slug to ID. Updates IncidentExport to map ID to slug

welshy557 committed:
Merge branch 'stats' of github.com:OsmiumCoder/rapid-report into stats

welshy557 committed:
Updates broken tests

LCschool committed:
Fixed tsconfig.json

LCschool committed:
Merge remote-tracking branch 'origin/stats' into stats

LCschool committed:
Split Graph into BarGraph.tsx and PieGraph.tsx, extracted footer method to GraphFooter.ts

LCschool committed:
Removed Witness count stat
Added location as stat
Added "Incident Lifetime"(days between creation and closed

LCschool committed:
Removed CSV button

OsmiumCoder committed:
small changes

OsmiumCoder committed:
resolved merge conflicts

OsmiumCoder committed:
fix code style
